sunkit_image/granule.py

```diff
@@ -0,0 +1,289 @@
+    else:
+        raise ValueError("Method must be one of: " + ", ".join(METHODS))
+    return threshold
+
+
+def trim_intergranules(segmented_image, mark=False):
+    """
+    Remove the erroneous identification of intergranule material in the middle
+    of granules that the pure threshold segmentation produces.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect extra intergranules.
+    mark : `bool`
+        If `False` (the default), remove erroneous intergranules.
+        If `True`, mark them as 0.5 instead (for later examination).
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndarray`
+        The segmented image without incorrect extra intergranules.
+    """
+    if len(np.unique(segmented_image)) > 2:
+        raise ValueError("segmented_image must only have values of 1 and 0.")
+    # Float conversion for correct region labeling.
+    segmented_image_fixed = np.copy(segmented_image).astype(float)
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    # Find value of the large continuous 0-valued region.
+    size = 0
+    for value in values:
+        if len((labeled_seg[labeled_seg == value])) > size:
+            real_IG_value = value
+            size = len(labeled_seg[labeled_seg == value])
+    # Set all other 0 regions to mark value (1 or 0.5).
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+    return segmented_image_fixed
+
+
+def mark_brightpoint(segmented_image, data, resolution):
+    """
+    Mark brightpoints separately from granules - give them a value of 1.5.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect middles.
+    data : `numpy array`
+        The original image.
+    resolution : `float`
+        Spatial resolution (arcsec/pixel) of the data.
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndrray`
+        The segmented image with brightpoints marked as 1.5.
+    brightpoint_count: `int`
+        The number of brightpoints identified in the image.
+    granule_count: `int`
+        The number of granules identified, after re-classifcation of brightpoint.
+    """
+    bp_size_limit = 0.1  # Approximate max size of a photosphere bright point in square arcsec (see Yanxiao et al., 2018)
+    bp_pix_upper_limit = bp_size_limit / (resolution**2)
+    bp_pix_lower_limit = 4 # Very small bright regions are likley artifacts
+    # General flux limit determined by visual inspection.
+    stand_devs = 0.5
+    bp_brightness_limit = np.mean(data) + stand_devs * np.std(data)

```

Comment:
Is this something we might want a user to pass in?

Reply:
That is a good idea